components/enterprise/subscriptions.tf

```diff
@@ -1,62 +1,37 @@
-  sds_non_production = { for k, v in var.sds_non_production_subscriptions : k => merge(tomap({ group = "sds_non_production" }), v) }
-  sds_production     = { for k, v in var.sds_production_subscriptions : k => merge(tomap({ group = "sds_production" }), v) }
-
-  crime = { for k, v in var.crime_subscriptions : k => merge(tomap({ group = "crime" }), v) }
-
-  heritage                = { for k, v in var.heritage_subscriptions : k => merge(tomap({ group = "heritage" }), v) }
-  heritage_sandbox        = { for k, v in var.heritage_sandbox_subscriptions : k => merge(tomap({ group = "heritage_sandbox" }), v) }
-  heritage_non_production = { for k, v in var.heritage_non_production_subscriptions : k => merge(tomap({ group = "heritage_non_production" }), v) }
-  heritage_production     = { for k, v in var.heritage_production_subscriptions : k => merge(tomap({ group = "heritage_production" }), v) }
-
-  security = { for k, v in var.security_subscriptions : k => merge(tomap({ group = "security" }), v) }
-
-  platform                = { for k, v in var.platform_subscriptions : k => merge(tomap({ group = "platform" }), v) }
-  platform_sandbox        = { for k, v in var.platform_sandbox_subscriptions : k => merge(tomap({ group = "platform_sandbox" }), v) }
-  platform_non_production = { for k, v in var.platform_non_production_subscriptions : k => merge(tomap({ group = "platform_non_production" }), v) }
-  platform_production     = { for k, v in var.platform_production_subscriptions : k => merge(tomap({ group = "platform_production" }), v) }
-
-  subscriptions = merge(
-    local.cft,
-    local.cft_sandbox,
-    local.cft_non_production,
-    local.cft_production,
-
-    local.sds,
-    local.sds_sandbox,
-    local.sds_non_production,
-    local.sds_production,
-
-    local.crime,
-
-    local.heritage,
-    local.heritage_sandbox,
-    local.heritage_non_production,
-    local.heritage_production,
-
-    local.security,
-
-    local.platform,
-    local.platform_sandbox,
-    local.platform_non_production,
-    local.platform_production,
-  )
-}
-
 module "subscription" {
   for_each = local.subscriptions
 
-  source = "../../modules/subscription"
-  name   = each.key
-  value  = each.value
+  source      = "../../modules/subscription"
+  name        = each.key
+  value       = each.value
+  common_tags = module.tags[each.key].common_tags
+  environment = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
 
   billing_account_name    = var.billing_account_name
   enrollment_account_name = var.enrollment_account_name
 }
+
+module "acme" {
+  for_each = { for k, v in local.subscriptions : k => v if try(v.deploy_acme, false) }
+
+  source = "git::https://github.com/hmcts/terraform-module-acme-function.git?ref=master"
+
+  location                       = var.location
+  env                            = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
```

Comment:
could the regex be extracted to a local so it's not repeated and it can be given a clear name explaining what it's doing?